tests/integration/test_manylinux.py

```diff
@@ -105,26 +107,69 @@ def io_folder(tmp_path):
     d.mkdir(exist_ok=True)
     return str(d)
 
+@contextmanager
+def tmp_docker_image(base, commands, setup_env={}):
+    """Make a temporary docker image for tests
+
+    Pulls the *base* image, runs *commands* inside it with *setup_env*, and
+    commits the result as a new image. The image is removed on exiting the
+    context.
+
+    Making temporary images like this avoids each test having to re-run the
+    same container setup steps.
+    """
+    client = docker.from_env()
+    logger.info("Pulling docker image %r", base)
```

Comment:
I wonder if we could just skip the pull step altogether. As far as I know the Docker daemon will pull the image anyway if it's not found locally, so in Travis we should always automatically get the latest version. For local testing this would mean faster startup if the image is already available locally. Even if the fixture is scoped to the session, repeatedly re-running a single test still incurs the cost of the pull every time the test is executed which adds up.

This would also enable testing with different versions of the image if needed, as various versions can be tagged locally for the test to pick up.

Reply:
I think that makes sense. I'll also add a note about it in the README.